taichi/system/timer.cpp

```diff
@@ -69,6 +69,24 @@ void Time::sleep(double s) {
   Time::usleep(s * 1e6_f64);
 }
 
+void Time::wait_until(double t) {
+  double dt;
+  if (t < Time::get_time()) {
+    return;
+  }
+  do {  // use system-provided sleep for large scale sleeping:
+    dt = (t - Time::get_time()) * 0.5;
+    if (dt <= 0) {
+      return;
+    }
+    Time::sleep(dt * 1e-3);
```

Comment:
We set the precision as 100us. But relatively , ```dt * 1e-3 ``` is too small . 
As a result , ```Sleep(0)``` is always called on windows , which will not give up the CPU time slice usually.
I think ```Time::sleep(dt)``` or ```Time::sleep(dt * 0.5)```  is enough.

Reply:
Note that the unit of `(t - Time::get_time())` is seconds, while the unit of `Time::sleep` is milliseconds.

Reply:
```Time::sleep``` should be seconds .  See ```Time::usleep(s * 1e6_f64)``` .